src/transaction/flashback.c

```diff
@@ -64,9 +64,99 @@ static volatile int flashback_Threshold_to_remove_archive = 0;	/* If the differe
 								 * the archive log volume can be deleted.
 								 */
 // *INDENT-OFF*
-static std::atomic_bool flashback_Is_active = false;	// the status value that the flashback is processing the request
+static std::atomic_bool flashback_Is_active = false;	// the status value that the flashback is processing the a request
+static std::atomic_bool flashback_In_progress = false;  // the status value that the flashback is in progress
 // *INDENT-ON*
 
+static CSS_CONN_ENTRY *flashback_Current_conn = NULL;	// the connection entry for a flashback requests
+
+static pthread_mutex_t flashback_Conn_lock = PTHREAD_MUTEX_INITIALIZER;
+
+/*
+ * flashback_is_duplicated_request - check if the caller is duplicated request for flashback
+ *
+ * return   : duplicated or not
+ */
+
+static bool
+flashback_is_duplicated_request (THREAD_ENTRY * thread_p)
+{
+  CSS_CONN_ENTRY *previous_conn = NULL;
+
+  /* When flashback is first called or exited properly,
+   * flashback_Current_conn is set to NULL */
+  if (flashback_Current_conn == NULL)
+    {
+      return false;
+    }
+
+  /* flashback_Current_conn indicates conn_entry in thread_p, and conn_entry can be reused by request handler.
+   * So, status in flashback_Current_conn can be overwritten. */
+
+  previous_conn = flashback_Current_conn;
+
+  if (flashback_In_progress == false)
+    {
+      /* previous flashback set flashback_In_progress to false prolperly. (exited well) */
+      return false;
+    }
+  else
+    {
+      /* flashback_In_progress == true */
```

Comment:
```suggestion
      /* flashback_In_progress == true */
      
      assert (previous_conn != NULL);
```

Reply:
It would be not bad to use just the flashback_Current_conn global variable.

Reply:
And I think that the flashback_Current_conn global variable can replace the purpose of the flashback_In_progress global variable. Then you can reduce/remove the usage of the global variable flashback_In_progress.

Reply:
I used `previous_conn` to make the code more understandable. 

do you mean that the condition in 'flashback_In_progress' can be changed to the condition in 'flashback_Current_conn' like below?
```
flashback_In_progress == false    ->  flashback_Current_conn == NULL
flashback_In_progress == true     ->  flashback_Current_conn != NULL 
```

Reply:
I understand what you mean.
I changed it as you said and tested it, and it worked well for the purpose.